syntax = "proto3";

package cart;

service CartService {
    rpc CreateCart(CreateCartRequest) returns (CreateCartResponse);
    rpc RemoveCart(RemoveCartRequest) returns (RemoveCartResponse);
    rpc AddToCart(AddToCartRequest) returns (AddToCartResponse);
    rpc RemoveFromCart(RemoveFromCartRequest) returns (RemoveFromCartResponse);
    rpc GetCart(GetCartRequest) returns (GetCartResponse);
}

message Cart {
    int32 id = 1;
    int32 user_id = 2;
}

message CartItem {
    int32 product_id = 1;
    int32 quantity = 2;
    float unit_price = 3;
}

message CreateCartRequest {
    int32 user_id = 1;
}

message CreateCartResponse {
    int32 cart_id = 1;
}

message RemoveCartRequest {
    int32 cart_id = 1;
}

message RemoveCartResponse {
    bool success = 1;
}

message AddToCartRequest {
    int32 cart_id = 1;
    int32 product_id = 2;
    int32 quantity = 3;
    float unit_price = 4;
}

message AddToCartResponse {
    bool success = 1;
}

message RemoveFromCartRequest {
    int32 cart_id = 1;
    int32 product_id = 2;
}

message RemoveFromCartResponse {
    bool success = 1;
}

message GetCartRequest {
    int32 cart_id = 1;
}

message GetCartResponse {
    Cart cart = 1;
    repeated CartItem items = 2;
    double total = 3;
}
